### PR TITLE
Adding chapter 8 exerises

### PR DESCRIPTION
**Pull request by pratyp**

### CODE REVIEW

File: chapter08/pxd/ex5.pl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+say "Enter a few lines to match words ending with 'a:";
+
+
+while (<>) { # take one input line at a time
+  chomp;
+  if (/ \b (?<word> \w+ a) \b
+  	# Perl 
+	# allows
+	# fancy
```
Comment (codavex) [inline chapter08/pxd/ex5.pl:16]: fancy

File: chapter08/pxd/ex1.pl [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use feature "say";
+
+say "Enter a few lines to match string 'match':";
```
Comment (codavex) [inline chapter08/pxd/ex1.pl:8]: This looks different to the pattern test program

File: chapter08/pxd/ex3.pl [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/perl
```
Comment (qianliok) [inline chapter08/pxd/ex3.pl:1]: This looks like the pattern test program

File: chapter08/pxd/ex3.pl [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+use feature "say";
+
+say "Enter a few lines to match any words ending with alphabet 'a':";
+
+while (<>) { # take one input line at a time
+  chomp;
+  if (/\b(\w*a)\b/i) {
+  	say "Matched: |$`<$&>$'|"; # the special match vars
+	say "\$1 contains " . $1;
```
Comment (qianliok) [inline chapter08/pxd/ex3.pl:14]: wondering what the `\` will do here?

File: chapter08/pxd/ex5.pl (hunk 1 above)
Comment (qianliok) [inline chapter08/pxd/ex5.pl:16]: cool, we can even have comments in pattern matching

File: chapter08/pxd/ex6.pl [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+use strict;
+use warnings;
+use feature "say";
+
+
+say "Enter a few lines ending with whitespace characters : ";
+
+while (<>) { # take one input line at a time
+  chomp;
+  if ( /[ \t]+$/ ) {
```
Comment (qianliok) [inline chapter08/pxd/ex6.pl:12]: how about `\s$`?

File: chapter08/pxd/ex3.pl (hunk 4 above)
Comment (codavex) [inline chapter08/pxd/ex3.pl:14]: It will escape the $, so you can print something like "$1 contains Wilma" instead of "Wilma contains Wilma"

**Pull request merged**